[REFACTOR & FIX] measurementView

DalioKim opened the pull request:
# 주요 변경사항
	- MeasurementView을 RunningRecordView로 네이밍 변경등 전체적인 네이밍 및 폴더경로 변경
	- AlertWhenTappedAndLongPressModifier 타입 추가 후 러닝 시작후 종료할 수 있는 기능 추가
	- ImageStyle 추가 및 IconStyle에 적용


# 중점적으로 봐줬으면 하는 부분
	- 네이밍
	- AlertWhenTappedAndLongPressModifier 로직
	- 테스트해야할 부분
	- 기록 종료 로직
		
# 예정 작업 사항
	- 추가된 기능의 TDD 구현
	- 리터럴을 변수로 변경 및 Swift Gen 적용
	- 이슈 버그 수정

[이슈 링크](https://github.com/ReleaseAppUntilDie/GatheRunner/issues/41)

DalioKim committed:
FEAT: stopButton 구현 (MeasurementView)

1. 커스텀 onLongPressGesture 기능 추상화한 TapAndLongPressWithAlertModifier 구현
2. ImageStyle파일 추가 및 asIconStyle 구현
3. MeasurementView에 시작버튼 클릭시 스톱버튼과 정지버튼으로 전환되게 구현

DalioKim committed:
REFACTOR:
1.폴더 경로, 파일, 메서드, 프로퍼티 네이밍 변경
2. ToggleStyle: IconStyle에 asIconStyle 적용 후 UI 로직 변경, size 프로퍼티 초기화에서 설정가능하게 변경

DalioKim committed:
REFACTOR: UITest 파일명 변경

wh5929 commented on GatheRunner/Helper/UI/Style/ImageStyle.swift:
```diff
@@ -0,0 +1,17 @@
+import SwiftUI
+
+extension Image {
+    func asIconStyle(withMaxWidth maxWidth: CGFloat = 20, withMaxHeight maxHeigh: CGFloat = 20) -> some View {
+        resizable()
+            .aspectRatio(contentMode: .fit)
+            .frame(width: maxWidth, height: maxHeigh)
+            .foregroundColor(.white)
```
### 오탈자
maxHeigh -> maxHeight 로 수정이 필요해보입니다.

DalioKim replied:
수정했습니다.

wh5929 commented on GatheRunner/Helper/UI/Style/Toggle+IconStyle.swift:
```diff
@@ -10,7 +10,7 @@ import SwiftUI
     let onImage: String
     let offImage: String
-    let size: CGFloat = 100
+    var size: CGFloat = 100
```
### 질문
상수에서 변수로 변경하신 이유가 궁금합니다.

DalioKim replied:
[답변] 상수로 선언시 .toggleStyle(IconStyle(onImage: "play", offImage: "pause", size: 100))와 같이 디폴트값을 변경할 수 가 없습니다. 디폴트 값과 변경 모두 가능하게 하기 위해 변수로 변경했습니다.

wh5929 commented on GatheRunner/Presentaion/Running/RunningRecordView.swift:
```diff
@@ -1,24 +1,23 @@
+// MARK: - RunningRecordView
 
-struct MeasurementView: View {
+struct RunningRecordView: View {
```
### 의견
MeasurementView 보다 RunningRecordView가 가독성이 더 좋네요.

DalioKim committed:
FIX: 파라미터 오타 수정

hanjaesueng commented on GatheRunner/Presentaion/Running/RunningRecordView.swift:
```diff
@@ -38,7 +37,7 @@ extension MeasurementView {
     }
 
-    var otherInfoView: some View {
+    var realTimeRecordView: some View {
         HStack(spacing: 30) {
             VStack(spacing: 10) {
                 Text(String(format: "%.2f", manager.distance))
```
질문
------------------
거리를 표시해주는 view 인거같은데 왜 realTimeRecord인가요??

DalioKim replied:
러닝중 실시간으로 기록을 측정해주는 화면이어서 기존 measurement에서 realTimeRecord라는 이름으로 변경했습니다.

hanjaesueng replied:
아 네 이해했습니다

DalioKim committed:
FIX: 스톱워치 버튼 아이콘 크기 문제 수정(이미지파일 변경 및 사이즈 크기 통일)

REFACTOR:
1. LongPressTypeAlertModifier, stopWatchButtonLayer등 네이밍 수정
2. 네임스페이스로 매직넘버, 매직문자 지정
3. UI 로직 수정(recordLabelView, resumeButton.onChange) 및 접근제한자 수정

DalioKim commented:
# 주요 변경사항
	- AlertWhenTappedAndLongPressModifier을 LongPressTypeAlertModifier 으로 변경 등, 타입 및 메서드 네이밍 변경
	- 네임스페이스로 매직넘버, 매직문자 지정
	- UI 로직 수정(recordLabelView, resumeButton.onChange) 및 접근제한자 수정
	- 스톱워치 부분 이슈 수정 

[이슈링크](https://github.com/ReleaseAppUntilDie/GatheRunner/issues/67)

# 중점적으로 봐줬으면 하는 부분
	- 네이밍
	- 매직넘버, 매직문자 네임스페이스로 지정할 필요가 있을지 현재 고민
	- Constats/SubViews 폴더 구조 및 파일명 적합여부
		
# 예정 작업 사항
	- 이미지 asset Swift Gen 적용 및 공통으로 이동

wh5929 commented on GatheRunner/Helper/UI/Modifier/LongPressTypeAlertModifier.swift:
```diff
@@ -0,0 +1,60 @@
+
+// MARK: - Constans
+
+private enum Constans {
```
### 의견

constants 의 오타인듯합니다

DalioKim replied:
오탈자 수정했습니다.

wh5929 commented on GatheRunner/Helper/UI/Modifier/LongPressTypeAlertModifier.swift:
```diff
@@ -0,0 +1,60 @@
+
+// MARK: - LongPressTypeAlertModifier
+
+struct LongPressTypeAlertModifier: ViewModifier {
+
+    @State var isAlertPresent = false
+
+    let longPressAction: () -> Void
+    let alertText: String
+    let alertDuration: Double
+    let LongPressDuration: Double
+
+    func body(content: Content) -> some View {
+        content
+            .onTapGesture {
+                $isAlertPresent.wrappedValue = true
+                Timer.scheduledTimer(withTimeInterval: alertDuration, repeats: false, block: { _ in
+                    $isAlertPresent.wrappedValue = false
+                })
+            }
+            .alert(isPresented: $isAlertPresent) {
+                Alert(title: Text(alertText), message: .none, dismissButton: .none)
+            }
+            .onLongPressGesture(minimumDuration: LongPressDuration) {
+                longPressAction()
+            }
+    }
+}
```
### 의견

이런 방식으로 event modifier 구현하면 되는거였군요. 참고하겠습니다.

wh5929 commented on GatheRunner/Helper/UI/Style/ImageStyle.swift:
```diff
@@ -0,0 +1,17 @@
+
+import SwiftUI
+
+extension Image {
+    func asIconStyle(withMaxWidth maxWidth: CGFloat = 20, withMaxHeight maxHeight: CGFloat = 20) -> some View {
+        resizable()
+            .aspectRatio(contentMode: .fit)
+            .frame(width: maxWidth, height: maxHeight)
+            .foregroundColor(.white)
+    }
```
### 질문

' . ' dot 연산자 없이 resizable() 을 사용했을때도 적용이 되네요.

'''''''''''''''''''''''''''
Content
    .resizable()
    .aspecRatio(contentMode: .fit)
'''''''''''''''''''''''''''
위의 코드와 같이 ViewModifier 없이 사용해도 되는 이유가 뭘까요?

DalioKim replied:
정확한건 아니지만, resizable() 옆에 self.가 생략된거고 self는 extension Image로부터 받아오기 때문인것 같습니다.

DalioKim committed:
FIX: 오탈자 수정

DalioKim committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

Conflicts:
	GatheRunner.xcodeproj/project.pbxproj
